superset-frontend/src/cccs-viz/plugins/plugin-chart-assembly-line/src/plugin/GenericComponent.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { styled } from '@superset-ui/core';
+import AssemblyLineRobot from '../images/Assemblyline_robot.png';
+
+const Container = styled.div`
+    border: 2px solid black;
+    height: 100px;
+`;
+
+const ResultImageContainer = styled.div`
+    padding-top: 10px;
+`;
+
+const Result = styled.strong`
+    padding-left: 10px;
+`;
+
+const Image = styled.img`
+    padding-right: 10px;
+    float: right;
+    height: 50px;
+    width: 50px;
+`;
+
+const ALLink = styled.a`
+    padding-left: 10px;
+    padding-bottom: 10px;
+`;
+
+interface GenericComponentProps {
+    totalTimesSeen: number;
+    ipAddress: string;
+}
+
+export default function GenericComponent(props: GenericComponentProps) {
```

Comment:
So I would like to see this a little more generic. So all AL based stuff (display string, link and image) should be passed in. that way if we want to reuse this for a different application we can fully resuse it

Reply:
You mean everything under one component without passing arguments?

Reply:
Not sure what you mean by "that" way.

Reply:
'that way' is in refrence to the previous sentence, so all AL based stuff (display string, link and image) should be passed in via the props. That way the generic component only handles rendering, it can the render the three things passed in

Reply:
Oh, I understand now. So the rendering should all be done in AssemblyLine() and then passed to the generic component.